kedro/contrib/io/dask/dask_parquet.py

```diff
@@ -0,0 +1,134 @@
+    remote data services to handle the corresponding load and save operations:
+        https://docs.dask.org/en/latest/remote-data-services.html
+
+        Example (AWS S3):
+        ::
+
+            >>> from kedro.contrib.io.dask.dask_parquet import DaskParquetDataSet
+            >>> import pandas as pd
+            >>> import dask.dataframe as dd
+            >>>
+            >>> data = pd.DataFrame({'col1': [1, 2], 'col2': [4, 5],
+            >>>                      'col3': [5, 6]})
+            >>> ddf = dd.from_pandas(data, npartitions=2)
+            >>>
+            >>> data_set = DaskParquetDataSet(
+            >>>     filepath="s3://bucket_name/path/to/folder",
+            >>>     credentials={
+            >>>         'client_kwargs':{
+            >>>             'aws_access_key_id': 'YOUR_KEY',
+            >>>             'aws_secret_access_key': 'YOUR SECRET',
+            >>>         }
+            >>>     },
+            >>>     save_args={"compression": "GZIP"}
+            >>> )
+            >>> data_set.save(ddf)
+            >>> reloaded = data_set.load()
+            >>>
+            >>> assert ddf.compute().equals(reloaded.compute())
+    """
+
+    DEFAULT_SAVE_ARGS = {"write_index": False}
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        filepath: str,
+        load_args: Dict[str, Any] = None,
+        save_args: Dict[str, Any] = None,
+        credentials: Dict[str, Any] = None,
+        fs_args: Dict[str, Any] = None,
+    ) -> None:
+        """Creates a new instance of ``DaskParquetDataSet`` pointing to concrete
+        parquet files.
+
+        Args:
+            filepath: Path to a parquet file
+                parquet collection or the directory of a multipart parquet.
+            load_args: Additional loading options `dask.dataframe.read_parquet`:
+                https://docs.dask.org/en/latest/dataframe-api.html#dask.dataframe.read_parquet
+            save_args: Additional saving options for `dask.dataframe.to_parquet`:
+                https://docs.dask.org/en/latest/dataframe-api.html#dask.dataframe.to_parquet
+            credentials: Credentials required to get access to the underlying filesystem.
+                E.g. for ``GCSFileSystem`` it should look like `{"token": None}`.
+            fs_args: Optional parameters to the backend file system driver:
+                https://docs.dask.org/en/latest/remote-data-services.html#optional-parameters
+        """
+        super().__init__(load_args, save_args)
+        self._filepath = filepath
+        self._fs_args = deepcopy(fs_args) or {}
+        self._credentials = deepcopy(credentials) or {}
+
+    @property
+    def fs_args(self) -> Dict[str, Any]:
+        """Property of optional file system parameters.
+
+        Returns:
+            A dictionary of backend file system parameters, including credentials.
+        """
+        fs_args = self._fs_args
```

Comment:
You should make an object copy here, otherwise you are updating `self._fs_args` with the contents of `self._credentials` too